### DISCUSSION TITLE
One Time Budget

### DISCUSSION THREAD

**Discussion by webtroter** [firefly-iii/firefly-iii#8185] [Firefly III] [answered]
Hello,

I'm probably misunderstanding how budget works, but I'm trying to understand how to set a non-recurring budget? Is there such a thing?

For example, I want to build a new computer, and I would like to budget this project some amount of money for all my purchases. Or that could also nicely apply to a travel budget : I want to take a vacation to Paris, and I'm budgeting 2000$ for the whole vacation.

While pondering about that, I had a thought : It would be nice if I could convert a piggy-bank into a one-time budget.
I save for a year for a vacation, then I convert the piggy bank into a one time budget when comes time to spend/break the piggy bank.

Also, it would be nice to have a selector on the budget page where I can choose which period view I want to see (week vs monthly) : I have many budget that are monthly, and a few others that are weekly. Not having to go back in my preference to change the period display would be nice.

**Comment by JC5** [comment, marked as answer]
It's funny how we went full circle on this: one time budgets used to be all we had. Just create a budget, set no auto-budget and fill in the amount on the `/budgets` page.

Auto-converting from a piggy bank to a (one time) budget will no be happening, that's on you to do.

For the page period switch, that's something I hope to add in the new layout.

**Reply by webtroter** [reply]
>  Just create a budget, set no auto-budget and fill in the amount on the /budgets page.

Then is there a reason why I see `This amount applies from November 1st, 2023 to November 30th, 2023:  ` on my one-time budget? Is it because of the date-range I use to view stuff?
I feel like a one-time budget shouldn't have amount for periods if the budget doesn't have one.

Also, when I click on a (one-time) budget, it brings me to the budget page for the current period. I feel like for those one-time budget I should get all transactions by default. Currently I manually remove the `period-id` from the URL.

**Reply by JC5** [reply]
Oh sorry no that's not possible. The budget amount always has a time range. Sorry.

**Reply by isaacgr** [reply]
So with no auto-budget, we have to re-enter the value each month?

Use case is similar to the above. I have a budget for a project. I want to set a fixed amount and subtract from that as I use it. I don't care about month-to-month costs per se. 

So I have 1000 budgeted for a basement reno. It might take several months to complete, I just want to subtract from that and track how much is left total. Ideally I don't want to have to go into the next month and update the budget to be what's left after the previous month.
